realm/src/androidTest/java/io/realm/DynamicRealmTest.java

```diff
@@ -0,0 +1,328 @@
+            typedRealm.cancelTransaction();
+        } finally {
+            typedRealm.close();
+            dynamicRealm.close();
+        }
+    }
+
+    // Test that Realms can only be deleted after all Typed and Dynamic instances are closed.
+    public void testDeleteAfterAllIsClosed() {
+        realm.close(); // Close Realm opened in setUp();
+        Realm typedRealm = Realm.getInstance(defaultConfig);
+        DynamicRealm dynamicRealm = DynamicRealm.getInstance(defaultConfig);
+
+        typedRealm.close();
+        try {
+            Realm.deleteRealm(defaultConfig);
+            fail();
+        } catch (IllegalStateException ignored) {
+        }
+
+        dynamicRealm.close();
+        assertTrue(Realm.deleteRealm(defaultConfig));
+    }
+
+    // Test that the closed Realm isn't kept in the Realm instance cache
+    public void testRealmCacheIsCleared() {
+        realm.close(); // Close DynamicRealm created in setup to prevent it interfering with caches here.
+        Realm typedRealm = Realm.getInstance(defaultConfig);
+        DynamicRealm dynamicRealm = DynamicRealm.getInstance(defaultConfig);
+
+        typedRealm.close(); // Still a dynamic instance open, but the typed Realm cache must still be cleared.
+        dynamicRealm.close();
+
+        try {
+            // If cache isn't cleared this would crash because of a closed shared group.
+            typedRealm = Realm.getInstance(defaultConfig);
+            assertEquals(0, typedRealm.where(AllTypes.class).count());
+        } finally {
+            typedRealm.close();
+        }
+    }
+
+    // Test that the closed DynamicRealms isn't kept in the DynamicRealm instance cache
+    public void testDynamicRealmCacheIsCleared() {
+        DynamicRealm dynamicRealm = DynamicRealm.getInstance(defaultConfig);
+        Realm typedRealm = Realm.getInstance(defaultConfig);
+
+        dynamicRealm.close(); // Still an instance open, but DynamicRealm cache must still be cleared.
+        typedRealm.close();
+
+        try {
+            // If cache isn't cleared this would crash because of a closed shared group.
+            dynamicRealm = DynamicRealm.getInstance(defaultConfig);
+            assertEquals(0, dynamicRealm.getVersion());
+        } finally {
+            dynamicRealm.close();
+        }
+    }
+
+    public void testCreateObject() {
+        realm.beginTransaction();
+        DynamicRealmObject obj = realm.createObject(CLASS_ALL_TYPES);
+        realm.commitTransaction();
+        assertTrue(obj.isValid());
+    }
+
+    public void testWhere() {
+        realm.beginTransaction();
+        realm.createObject(CLASS_ALL_TYPES);
+        realm.commitTransaction();
+
+        RealmResults<DynamicRealmObject> results = realm.where(CLASS_ALL_TYPES).findAll();
+        assertEquals(1, results.size());
+    }
+
+    public void testClearInvalidName() {
+        realm.beginTransaction();
+        realm.clear("I don't exist");
```

Comment:
This doesn't fail? Strange, `BaseRealm.getTable()` should return a `null` and calling `clear` on `Table null` should throw, right?